printTables.py

```diff
@@ -0,0 +1,247 @@
+
+
+
+"""
+   Discord has a max width of 80 characters until it word wraps. Therefore, 79 is the max word length
+   Create dynamic column sizes by creating an array for width of each column.
+
+   make maxWordLen =  max row size
+
+
+"""
+class OutputTable:
+
+
+    def __init__(self, matrixTable, maxRowWidth):
+        self.table = matrixTable
+        self.boarder = "|"
+        self.rowSym = "-"
+        self.intersectionChar = "+"
+        self.outputString = ""
+        self.tableSize = len(self.table)
+        self.headerSize = 0
+        self.getHeaderSize()
+        self.maxWordLen = int(maxRowWidth / self.headerSize)
+        self.maxRowWidth = maxRowWidth
+        self.columnWidth = []
+
+
+
+
+
+    def getHeaderSize(self):
+
+        if self.tableSize > 0:
+            self.headerSize = len(self.table[0])
+
+
+    """
+    Accessor function for the maximum word length variable 
+    """
+    def getMaxWordLength(self):
+
+        return self.maxWordLen
+
+    """
+    Mutator function maximum word length  variable 
+    """
+    def setMaxWordLength(self, newLen):
+
+        self.maxWordLen = newLen
+
+    """
+     Accessor function for the boarder character variable
+    """
+    def getBoarderChar(self):
+
+        return str(self.boarder)
+
+    """
+    Mutator function for the boarder character variable 
+    """
+    def setBoarderChar(self, newChar):
+
+        self.boarder = str(newChar)
+
+    def makeString(self, appendString):
+
+        self.outputString += appendString
+
+    """
+     Function is designed to multiply the member variable self.boarder to a specified number 
+     This function helps create the character designed boarder
+    """
+    def mulWordWithrowChar(self, num):
+
+        return str(self.rowSym * num)
+
+    """
+    Function calculates the distance for where the last self.boarder  needs to be placed when printing a row. 
+    """
+    def finalSymError (self):
+
+        return self.maxWordLen - 18
+
+    """
+    Function calculates the number of self.boarder chars needed to align perfectly, when printing a solid lin 
+    of all self.boarder chars. This is useful when the self.maxWordLen changes and the table boarders need to be 
+    resized. 
+    """
+    def resizing (self):
+
+        return self.maxWordLen * 5 - 3
```

Comment:
Why is this * 5 - 3? Is 5 the number of columns?

Reply:
This math also was an illusion. Seemed to work initially, but with data that is not perfectly  symmetrical it fails.